parsers/lib/session.py

```diff
@@ -0,0 +1,19 @@
+import ssl
+
+import urllib3
+from requests import Session, adapters
+
+
+class LegacyHttpAdapter(adapters.HTTPAdapter):
+    def init_poolmanager(self, connections, maxsize, block=False):
+        ctx = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+        ctx.options |= 0x4  # OP_LEGACY_SERVER_CONNECT
+        self.poolmanager = urllib3.poolmanager.PoolManager(
+            num_pools=connections, maxsize=maxsize, block=block, ssl_context=ctx
+        )
+
+
+# Use a LegacyHttpAdapter to avoid "unsafe legacy renegotiation disabled" error
+# Original code source: https://stackoverflow.com/questions/71603314/ssl-error-unsafe-legacy-renegotiation-disabled
+def mount_legacy_adapter(session: Session):
+    session.mount("https://", LegacyHttpAdapter())
```

Comment:
This works great for the current parser but to make it a little more robust I suggest we change it to something like this instead:
```suggestion
def get_legacy_adapter(session: Session, logger: Logger):
    local_session = deepcopy(session)
    if "https://" in local_session.adapters or "http://" in local_session.adapters:
        logger.warning("Session already has adapters, they will be overridden.")
    local_session.mount("https://", LegacyHttpAdapter())
    local_session.mount("http://", LegacyHttpAdapter())
    return local_session
```
This adopts a functional style for the function to avoid poluting the original session object with the changes.

I double checked in the backend and a new session is only created per *parser* not per parser function. For most parsers this won't be an issue as all the functions call the same domain. But a few parsers have diffrent domains for say production and consumption. The above changes make sure we only change it in the parser function that is called.

Reply:
Another option to the above would be to create a new session in this function instead, like this:
```suggestion
def get_session_with_legacy_adapter():
	    session = Session();
	    session.mount("https://", LegacyHttpAdapter())
	    session.mount("http://", LegacyHttpAdapter())
	    return session
```
It would simplify things a bit.